src/main/java/edu/harvard/iq/dataverse/datasetutility/AddReplaceFileHelper.java

```diff
@@ -610,11 +611,14 @@ private boolean runAddReplacePhase1(Dataset owner,
             return false;
             
         }
-        if(optionalFileParams != null) {
-        	if(optionalFileParams.hasCheckSum()) {
-        		newCheckSum = optionalFileParams.getCheckSum();
-        		newCheckSumType = optionalFileParams.getCheckSumType();
-        	}
+        if (optionalFileParams != null) {
+            if (optionalFileParams.hasCheckSum()) {
+                newCheckSum = optionalFileParams.getCheckSum();
+                newCheckSumType = optionalFileParams.getCheckSumType();
+            }
+            if (optionalFileParams.hasFileSize()) {
```

Comment:
This is going to make fileSize a valid param for direct uploads, etc. now too? I've been hesitant to trust the client about this (can I say my files are 10 bytes and get a lot of free storage?). Would it make sense to allow the Globus code to set this, e.g. passing a boolean through the addFiles call on #1092 that determines whether file size is read here? Not sure we need this or that this is the best way, but thought I'd raise the question.

Reply:
That's fine with me. To confirm: we'll be allowing Globus to do this, internally; but not accepting these entries in the json passed to `/addFiles`. 

Also (somewhat unrelated) it would be fairly easy to implement a very similar mass lookup on an s3 folder inside an `/addFiles` call finalizing a large batch of direct s3 uploads. *If* it is ever registered to be a bottleneck. I never got around to measuring just how much these individual s3 lookups cost, but it always bothered me a little bit that we make separate network/http calls for them. (they definitely do not cost anything remotely approaching what I got with globus and NESE though)